### ISSUE TITLE
CSS is not Regenerated if Page/Template uses PHP Function Containing $global Product Classes

### ISSUE THREAD

**Issue by Ulprus** [soflyy/oxygen-bugs-and-features#3103]
**Describe the bug**
When setting up a PHP function in a snippet program (IE, Advanced Scripts) which echoes or returns a Product Class of WooCommerce (example: `$product->get_sku();`), whichever page or template uses this PHP Function will cause an AJAX error when regenerating that page's/template's CSS cache.

**A link to a Sandbox site where the bug has been reproduced https://oxygenbuilder.com/try? (REQUIRED)**
Link: [Admin Login](https://adorablecrane.build067.wpsandbox.app/wp-admin/)
Username: TestUser
Password: TestUserPass123!@#

**IF YOU CAN'T PROVIDE A SANDBOX REPRODUCTION: Steps required to reproduce the bug:**
1. Install WooCommerce and add a product along with an SKU.
2. Set up a PHP function in snippet software (in this case, I used Advanced Scripts):

```
<?php

/* This function breaks Oxygen's 'Regenerate CSS' for some reason */

function u_product_sku()
{

    global $product;
    $productSKU = $product->get_sku();

    echo $productSKU;
}
```

3. Set up a template for post_type `product`
4. Add a Repeater with a default query
5. In the repeater, convert a text block to dynamic data
6. Choose Return PHP Function
7. Use `u_product_sku`
8. The function works well and as intended in both the builder and the front.
9. Now in Oxygen's settings, click on Regenerate CSS Cache.
10. There will be an AJAX error for this template.

**Comment by Kpudlo** [comment]
Hello @Ulprus, 

This was an issue that should have been fixed in Oxygen 4.0.4 when PHP such as `global $product` is used in a Code Block element.

I wasn't able to see the issue on the Sandbox site, but when using an external script, you may want to wrap the code within an if statement to see if that fixes the issue: 

```php
<?php

/* This function breaks Oxygen's 'Regenerate CSS' for some reason */

function u_product_sku()
{

    global $product;
    if($product) {
         $productSKU = $product->get_sku();

         echo $productSKU;
    }
}
```

**Comment by Ulprus** [comment]
Hi @Kpudlo - apologies for the delayed response; email never went into my primary inbox.

That worked perfectly! Thanks so much. Really appreciate this.
